divi/src/net.cpp

```diff
@@ -1454,6 +1460,16 @@ void ThreadMessageHandler()
                     g_signals.SendMessages(pnode, pnode == pnodeTrickle || pnode->fWhitelisted);
             }
             boost::this_thread::interruption_point();
+
+            if (rebroadcast) {
+                // Periodically clear setAddrKnown to allow refresh broadcasts
+                if (nLastRebroadcast)
+                    pnode->setAddrKnown.clear();
+
+                // Rebroadcast our address
+                AdvertizeLocal(pnode);
+                boost::this_thread::interruption_point();
+            }
```

Comment:
In order to preserve behaviour as much as possible, while avoiding the deadlock, wouldn't it be preferable to insert this added codeblock just before the line `g_signals.SendMessages(pnode, pnode == pnodeTrickle || pnode->fWhitelisted);`

Reply:
Or maybe even before the loop over CNode* 's

Reply:
Ok, so the invocation of g_signals.SendMessages has a node-specific ping/pong check. This can be extracted into a function on CNode directly (e.g. called HasSyncedPingPongMessages() ). If you could inlcude that as well in the PR, that would very helpful

Reply:
This change is not meant to be 100% the same as what the code before was, and is also a cleanup in some sense.  It just makes no sense to have a loop over all nodes and doing that rebroadcasting thing inside the node-specific `SendMessages`.  (This effectively executed the rebroadcast after the ping/pong logic of the first node, but before all other nodes, which seems just arbitrary.)

I don't see a reason to specifically try to reproduce the exact structure of before, rather than just use the simpler structure it has now.  Or are you aware of anything specific why it needs to be done as it was before?